### PR TITLE
Fgts 119 create rat enemy

### PR DESCRIPTION
**Pull request by jDenyes**
Added the enemy rat in

### CODE REVIEW

**Comment by fheysel** [comment]
I pulled the branch and put sent some comments in the Teams chat along with a video but I'll post them here too for reference. 

The rat enemy is looking good but there are still a few things that need to change before its demo ready. The first is that we still need to jump into frame for the enemy to see us. The second is that even when you jump into frame you have to be standing very close to get seen by the rat, the raycasts can be extended so that his jump ends right at the player. The last thing is that the hit box is too large. When you watch the video you can see that the player dies far before the rat ever actually touches them. It would be pretty frustrating from a players pov

File: godotproject/rat_spritesheet.png.import [hunk 1]
```diff
@@ -2,15 +2,15 @@
 
 importer="texture"
 type="StreamTexture"
-path="res://.import/birds.png-e63708046b8c22ce4989dac9da8485df.stex"
+path="res://.import/rat_spritesheet.png-0ef4b8f8dd66edc1ec1cb2e54751a5e7.stex"
```
Comment (charasyn) [inline godotproject/rat_spritesheet.png.import]: Git seems to have marked this as replacing the `birds.png.import` file, which I don't think is correct and will probably break the bird. Pls give it another check as to whether this is intended and whether everything still works

**Review by charasyn — CHANGES_REQUESTED** [review]

**Comment by fheysel** [comment]
This looks good for the most part, I pulled it and played but there are a few small thing. 1 lower the rats z index so the player is always in front. 2. rat attack deals 0 damage atm, increase that. 3. Something is printing a lot to std out, remove print statements. 4. The rat should have sound effects for when it sees the player and then when it attacks, something like a growl and then a slash sound or something. Check out https://freesound.org/ thats where we've been getting all our sound effects from.

**Comment by fheysel** [comment]
The rat also does not take damage

**Review by fheysel — APPROVED** [review]
Changes look great! I love the sound effects you chose :)

**Review by charasyn — APPROVED** [review]

**Review by charasyn — CHANGES_REQUESTED** [review]
Wait hold on

**Pull request merged**